ote/src/cljc/ote/transit_changes.cljc

```diff
@@ -138,6 +138,39 @@
 (defn earliest-stop-sequence [stop]
   (:gtfs/stop-sequence stop))
 
+(defn sequence-for-stop [stop stops]
+  (let [date1-departure-time (:gtfs/departure-time-date1 stop)
+        reversed-stops (reverse stops)
+        modified-stop (some
+                                 (fn [s]
+                                   (let [sequence (:gtfs/stop-sequence s)]
+                                     (when (< (time/interval->seconds (:gtfs/departure-time-date1 s)) (time/interval->seconds date1-departure-time))
+                                       (assoc stop :gtfs/stop-sequence (* sequence 1.01))))) ; create new sequence nro for removed stop
+                                 reversed-stops)]
+    (if (or (empty? modified-stop) (nil? modified-stop))
+      stop
+      modified-stop)))
+
+(defn reorder-removed-stops
```

Comment:
Would following simplification work?

- set following as inititial `val` (https://clojuredocs.org/clojure.core/reduce) of reduce:
```
(filterv #(not (nil? (:gtfs/departure-time-date2 %))) indexed-stops)
``` 
- in reduce return 
```
(conj result (sequence-for-stop stop result))
```

Reply:
+ sort-by to conj result

Reply:
Not sure if I want make it more compact. It is easier and faster to read and understand when things are stored in named variables. But I'm not sure if those names are that good :)

Reply:
a) No problem on documenting by using intermediate symbols with good names
(though one school of thought e.g. in "Elements of Clojure" is that if once can't come up with a good name one should a) sleep on it or b) not use any :) )

b) You might have missed the key point I was proposing: having `reduce` act as a function and using _only_ symbols which are given to it as arguments.
Now it's using ` other-stops` outside the reduce context.
It would me more maintainable and simple if ` other-stops` is given as the initial value 
and reduce iteration conjoins the added stop into it and returns the result for next iteration,
which does the same again.
:)

Reply:
That will remove one `if` in the reduce. And actually the `if` shouldn't be necessary: `conj coll nil` returns coll so it's safe to conjoin nil.